index.js

```diff
@@ -80,45 +77,29 @@ function decode (buffer) {
  *  62300 => 0x00f35c
  * -62300 => 0xff0ca4
 */
-function encode (r, s) {
+exports.encode = function encode (r, s) {
   var lenR = r.length
-  if (r[0] & 0x80) lenR += 1
-  if (lenR > 33) throw new Error('R length is too long')
-
   var lenS = s.length
-  if (s[0] & 0x80) lenS += 1
-  if (lenS > 33) throw new Error('S length is too long')
+  if (lenR === 0) throw new Error('R length is zero')
+  if (lenS === 0) throw new Error('S length is zero')
+  if (r[0] & 0x80) throw new Error('R value is negative')
+  if (s[0] & 0x80) throw new Error('S value is negative')
+  if (r[0] === 0x00 && !(r[1] & 0x80)) throw new Error('R value excessively padded')
+  if (s[0] === 0x00 && !(s[1] & 0x80)) throw new Error('S value excessively padded')
+  if (lenR > 33 || (lenR > 32 && r[0] !== 0x00)) throw new Error('R length is too long')
+  if (lenS > 33 || (lenS > 32 && s[0] !== 0x00)) throw new Error('S length is too long')
 
   var signature = new Buffer(6 + lenR + lenS)
 
   // 0x30 [total-length] 0x02 [R-length] [R] 0x02 [S-length] [S]
   signature[0] = 0x30
   signature[1] = signature.length - 2
   signature[2] = 0x02
-  signature[3] = lenR
-
-  if (r[0] & 0x80) {
-    signature[4] = 0x00
-    r.copy(signature, 5)
-  } else {
-    r.copy(signature, 4)
-  }
-
+  signature[3] = r.length
+  r.copy(signature, 4)
   signature[4 + lenR] = 0x02
-  signature[5 + lenR] = lenS
-
-  if (s[0] & 0x80) {
-    signature[6 + lenR] = 0x00
-    s.copy(signature, 7 + lenR)
-  } else {
-    s.copy(signature, 6 + lenR)
-  }
+  signature[5 + lenR] = s.length
+  s.copy(signature, 6 + lenR)
 
   return signature
 }
-
-module.exports = {
-  check: check,
-  decode: decode,
-  encode: encode
-}
```

Comment:
@fanatid can't stand it.
I follow these three rules which lead to the result I typically have:

* No use before declaration
* No repeated declarations
* Prefer named functions
* Group hackish/lame/ugly code for easy removal one day

Moving the exports to the top breaks my first rule.
The current implementation breaks my second rule,  and removing the function name makes it an anonymous function (ew).
The previous implementation met the first two rules,  and adheres to third.

Ideally,  ES Next's `export`  style syntax would be available... until then,  I use the `module.exports = {}` at EOF.

Thoughts?